src/index.ts

```diff
@@ -60,28 +60,48 @@ async function verifyKey(
   timestamp: string,
   clientPublicKey: string,
 ): Promise<boolean> {
-  return await edVerify(signature, Buffer.concat([Buffer.from(timestamp, 'utf-8'), rawBody]), clientPublicKey);
+  try {
+    return await edVerify(signature, Buffer.concat([Buffer.from(timestamp, 'utf-8'), rawBody]), clientPublicKey);
+  } catch (e) {
+    // In case of failure, assume verification failure and return false.
+    //
+    // The Point#multiply error from noble-ed25519 comes from a bad signature which cannot be parsed correctly
+    // by the library. It's just a poor error handling case on the library's end which causes a vague error.
+    // - LC
+    return false;
+  }
 }
 
+type NextFunction = (err?: Error) => void;
+
 /**
  * Creates a middleware function for use in Express-compatible web servers.
  *
  * @param clientPublicKey - The public key from the Discord developer dashboard
  * @returns The middleware function
  */
-function verifyKeyMiddleware(clientPublicKey: string): (req: Request, res: Response, next: NextFunction) => void {
+function verifyKeyMiddleware(
+  clientPublicKey: string,
+): (req: IncomingMessage, res: ServerResponse, next: NextFunction) => void {
   if (!clientPublicKey) {
     throw new Error('You must specify a Discord client public key');
   }
-  return async function (req: Request, res: Response, next: NextFunction) {
-    const timestamp = (req.header('X-Signature-Timestamp') || '') as string;
-    const signature = (req.header('X-Signature-Ed25519') || '') as string;
+  return async function (req: IncomingMessage, res: ServerResponse, next: NextFunction) {
+    const timestamp = (req.headers['x-signature-timestamp'] || '') as string;
+    const signature = (req.headers['x-signature-ed25519'] || '') as string;
 
     const chunks: Array<Buffer> = [];
     req.on('data', function (chunk) {
       chunks.push(chunk);
     });
     req.on('end', async function () {
+      // header sanity check
+      if (!timestamp || !signature) {
+        res.statusCode = 401;
+        res.end('Invalid headers');
+        return;
+      }
+
```

Comment:
Let's move this to the top of the function, and we can also get rid of the `|| ''`